File: app/src/main/java/com/rollapp/shared/ui/onboarding/OnboardingScreen.kt
package com.rollapp.shared.ui.onboarding

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.AutoAwesome
import androidx.compose.material.icons.rounded.PhotoCamera
import androidx.compose.material.icons.rounded.PhotoLibrary
import androidx.compose.material3.Button
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import kotlinx.coroutines.launch
import androidx.compose.foundation.border
import com.rollapp.shared.ui.components.GoldButton
import com.rollapp.shared.ui.components.Hairline
import com.rollapp.shared.ui.theme.Gold
import com.rollapp.shared.ui.theme.Raised

private data class OnboardingPage(
    val icon: ImageVector,
    val title: String,
    val body: String
)

private val pages = listOf(
    OnboardingPage(
        Icons.Rounded.PhotoCamera,
        "Capture together",
        "Take photos straight from WeWere and they land in the roll instantly."
    ),
    OnboardingPage(
        Icons.Rounded.PhotoLibrary,
        "One shared camera roll",
        "Everyone's photos gather in one place, so nobody has to ask for the good ones."
    ),
    OnboardingPage(
        Icons.Rounded.AutoAwesome,
        "Keep the memories",
        "Your roll is a private collection of the whole trip. Nobody else can see it."
    )
)

@Composable
fun OnboardingScreen(onFinished: () -> Unit) {
    val pagerState = rememberPagerState(pageCount = { pages.size })
    val scope = rememberCoroutineScope()
    val isLast = pagerState.currentPage == pages.lastIndex

    // The activity draws edge to edge, so this screen owns its own insets — without
    // them Skip sits underneath the clock and the battery icon.
    Column(
        modifier = Modifier
            .fillMaxSize()
            .statusBarsPadding()
            .navigationBarsPadding()
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 8.dp, vertical = 8.dp),
            horizontalArrangement = Arrangement.End
        ) {
            TextButton(onClick = onFinished) { Text("Skip") }
        }

        HorizontalPager(
            state = pagerState,
            modifier = Modifier.weight(1f)
        ) { index ->
            val page = pages[index]
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(horizontal = 36.dp),
                verticalArrangement = Arrangement.Center,
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Box(
                    modifier = Modifier
                        .size(112.dp)
                        .clip(CircleShape)
                        .border(1.dp, Gold.copy(alpha = 0.6f), CircleShape)
                        .background(Raised),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = page.icon,
                        contentDescription = null,
                        tint = Gold,
                        modifier = Modifier.size(44.dp)
                    )
                }
                Spacer(Modifier.height(36.dp))
                Text(
                    text = page.title,
                    style = MaterialTheme.typography.headlineLarge,
                    textAlign = TextAlign.Center
                )
                Spacer(Modifier.height(12.dp))
                Hairline(Modifier.width(120.dp))
                Spacer(Modifier.height(12.dp))
                Text(
                    text = page.body,
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    textAlign = TextAlign.Center
                )
            }
        }

        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 24.dp),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically
        ) {
            repeat(pages.size) { index ->
                val selected = pagerState.currentPage == index
                val width by animateFloatAsState(
                    targetValue = if (selected) 22f else 8f,
                    label = "indicator"
                )
                Box(
                    modifier = Modifier
                        .padding(horizontal = 3.dp)
                        .height(8.dp)
                        .width(width.dp)
                        .clip(CircleShape)
                        .background(
                            if (selected) MaterialTheme.colorScheme.primary
                            else MaterialTheme.colorScheme.outlineVariant
                        )
                )
            }
        }

        GoldButton(
            text = if (isLast) "Get started" else "Next",
            onClick = {
                if (isLast) onFinished()
                else scope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) }
            },
            modifier = Modifier.padding(horizontal = 28.dp)
        )

        Spacer(Modifier.height(32.dp))
    }
}

File: ios/WeWere/UI/Screens/OnboardingScreen.swift
import SwiftUI

private struct OnboardingPage {
    let icon: String
    let title: String
    let body: String
}

private let pages: [OnboardingPage] = [
    .init(icon: RollIcon.camera, title: "Capture together",
          body: "Take photos straight from WeWere and they land in the roll instantly."),
    .init(icon: RollIcon.photoLibrary, title: "One shared camera roll",
          body: "Everyone's photos gather in one place, so nobody has to ask for the good ones."),
    .init(icon: RollIcon.autoAwesome, title: "Keep the memories",
          body: "Your roll is a private collection of the whole trip. Nobody else can see it.")
]

struct OnboardingScreen: View {
    let onFinished: () -> Void

    @State private var page = 0

    private var isLast: Bool { page == pages.count - 1 }

    var body: some View {
        VStack(spacing: 0) {
            HStack {
                Spacer()
                TextButton(text: "Skip", action: onFinished)
            }
            .padding(.horizontal, 8)
            .padding(.vertical, 8)

            TabView(selection: $page) {
                ForEach(Array(pages.enumerated()), id: \.offset) { index, item in
                    VStack(spacing: 0) {
                        ZStack {
                            Circle().fill(Raised)
                            Circle().stroke(Gold.opacity(0.6), lineWidth: 1)
                            Image(systemName: item.icon)
                                .font(.system(size: 40, weight: .medium))
                                .foregroundStyle(Gold)
                        }
                        .frame(width: 112, height: 112)
                        Spacer().frame(height: 36)
                        Text(item.title)
                            .rollStyle(RollType.headlineLarge, color: Ivory)
                            .multilineTextAlignment(.center)
                        Spacer().frame(height: 12)
                        HairlineRule().frame(width: 120)
                        Spacer().frame(height: 12)
                        Text(item.body)
                            .rollStyle(RollType.bodyLarge, color: IvoryMuted)
                            .multilineTextAlignment(.center)
                    }
                    .padding(.horizontal, 36)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                    .tag(index)
                }
            }
            .tabViewStyle(.page(indexDisplayMode: .never))

            HStack(spacing: 6) {
                ForEach(0..<pages.count, id: \.self) { index in
                    Capsule()
                        .fill(page == index ? Gold : OutlineVariant)
                        .frame(width: page == index ? 22 : 8, height: 8)
                        .animation(.easeOut(duration: 0.25), value: page)
                }
            }
            .padding(.bottom, 24)

            GoldButton(text: isLast ? "Get started" : "Next") {
                if isLast { onFinished() } else { withAnimation { page += 1 } }
            }
            .padding(.horizontal, 28)

            Spacer().frame(height: 32)
        }
        .background(Ink.ignoresSafeArea())
    }
}
